src/keycloak/keycloak_admin.py

```diff
@@ -214,146 +215,254 @@ def connection(self, value):
+        current_version=__version__,
+        details="Use the connection.totp property instead",
+    )
     def totp(self, value):
-        self._totp = value
+        self.connection.totp = value
 
     @property
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.token property instead",
+    )
     def token(self):
         """Get token.
 
         :returns: Access and refresh token
         :rtype: dict
         """
-        return self._token
+        return self.connection.token
 
     @token.setter
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.token property instead",
+    )
     def token(self, value):
-        self._token = value
-
-    @property
-    def auto_refresh_token(self):
-        """Get auto refresh token.
-
-        :returns: List of methods for automatic token refresh
-        :rtype: list
-        """
-        return self._auto_refresh_token
+        self.connection.token = value
 
     @property
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.user_realm_name property instead",
+    )
     def user_realm_name(self):
         """Get user realm name.
 
         :returns: User realm name
         :rtype: str
         """
-        return self._user_realm_name
+        return self.connection.user_realm_name
 
     @user_realm_name.setter
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.user_realm_name property instead",
+    )
     def user_realm_name(self, value):
-        self._user_realm_name = value
+        self.connection.user_realm_name = value
 
     @property
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.custom_headers property instead",
+    )
     def custom_headers(self):
         """Get custom headers.
 
         :returns: Custom headers
         :rtype: dict
         """
-        return self._custom_headers
+        return self.connection.custom_headers
 
     @custom_headers.setter
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.custom_headers property instead",
+    )
     def custom_headers(self, value):
-        self._custom_headers = value
+        self.connection.custom_headers = value
+
+    @property
+    @deprecation.deprecated(
+        deprecated_in="2.13.0",
+        removed_in="3.0.0",
+        current_version=__version__,
+        details="Use the connection.auto_refresh_token property instead",
```

Comment:
This is not correct, right? The `connection` does not have the `auto_refresh_token` attribute. Maybe a message like `"Auto-refresh will be implicitly set for all requests"` instead?

Reply:
Thanks for catching that! Fixed.